config/env_spec_request.go

```diff
@@ -423,11 +445,29 @@ func (e *EnvironmentSpecRequest) verifyJWTAuthentication(name string) error {
 		}
 	}
 
-	return fault.NewAdapterFaultWithRpcCode(rpc.UNAUTHENTICATED)
+	return adapterFaultForJwtErr(err)
+}
+
+func adapterFaultForJwtErr(err error) *fault.AdapterFault {
+	if err == nil {
+		return fault.NewAdapterFault(fault.JwtUnknownException, rpc.UNAUTHENTICATED, 0)
+	}
+
+	if errors.Is(err, ErrUnsupportedJwkSource) {
```

Comment:
Might this be easier to read as a single switch/case statement?

Reply:
thanks for the suggestion, replaced with switch.